### PR TITLE
Ft create test api db 162123750

### PR DESCRIPTION
**Pull request by bobbylemm**
WHAT DOES THIS PR DO?
implements all the feedback on the previous PR.

WHAT ARE THE RELEVANT PT STORIES ID?
#162123750

### CODE REVIEW

File: Server/controllers/ParcelController.js [hunk 1]
```diff
@@ -30,49 +30,60 @@ class ParcelController {
...
+        return e;
+    }
+}
+
 // this is the controller to update the status of a parcel
 static async updateParcelStatus (req, res) {
     const { pid } = req.params;
     const { newStatus } = req.body;
-    const validStatus = ['in-transit', 'delivered', 'cancelled'].includes(newStatus);
+    const validStatus = ['in-transit', 'delivered'].includes(newStatus);
```
Comment (saheedt) [inline Server/controllers/ParcelController.js:89]: For clarification, is `cancelled` not a valid option? I believe the user should be able to cancel the send order.

File: Server/routes/route.js [hunk 2]
```diff
@@ -1,46 +1,42 @@
...
 // this is the route to get all parcels for a user
-// GET ALL PARCELS
-router.get('/parcels/user', parcelController.getParcelsByUser);
-// this is the route to get a specific parcel
-// GET A SPECIFIC PARCEL
-// router.get('/parcels/:id', parcelController.getSpecificParcel);
-// this is to change the dropofflocation of a parcel
-// PUT IN A NEW DROPOFFLOCATION
-router.put('/parcels/:pid/newlocation', validateToken, validateAdmin, parcelController.updateParcelPresentLocation);
+router.get('/parcels/user',validateToken, getParcelsByUser);
```
Comment (saheedt) [inline Server/routes/route.js]: This route is wrong consult your rubric for the right route.
Also, the controller to this route should not be in the the `ParcelController`  but `UsersController`

File: Server/controllers/ParcelController.js (hunk 1 above)
Comment (bobbylemm) [reply Server/controllers/ParcelController.js:89]: no this is not a route for the user, this is accessible to the admin.
the user has a route that handles the cancel request.

Comment (saheedt) [reply Server/controllers/ParcelController.js:89]: ok cool

File: Server/routes/route.js [hunk 3]
```diff
@@ -1,46 +1,42 @@
...
 // fetch all parcels in the application
 // ------------------------admin only------------------
 // GET ALL PARCELS IN THE APP (accessible to admin only)
-router.get('/parcels', validateToken, validateAdmin, parcelController.getAllParcels);
+router.get('/parcels/users', validateToken, validateAdmin, getAllParcels);
+// this is to get all parcels for a specific user
+router.get('/parcels/:uid', validateToken, validateAdmin, getAllParcelsBySpecificUser)
 // this is the route for an admin to update the status of a parcel delivery order
-router.put('/parcels/:pid/status', validateToken, validateAdmin, parcelController.updateParcelStatus);
+router.patch('/parcels/:pid/status', validateToken, validateAdmin, updateParcelStatus);
```
Comment (saheedt) [inline Server/routes/route.js]: this is supposed to be a `PUT` and not a `PATCH`. Are you sure you are looking at the rubrics?

File: Server/routes/route.js [hunk 4]
```diff
@@ -1,46 +1,42 @@
...
+// this is to change the dropofflocation of a parcel
+router.patch('/parcels/:pid/currentlocation', validateToken, validateAdmin, updateParcelPresentLocation);
 // this is for the superadmin
-router.put('/superadmin/createadmin', validateSuperAdmin, usersControllers.createAdmin);
+router.patch('/superadmin/createadmin', validateSuperAdmin, createAdmin);
```
Comment (saheedt) [inline Server/routes/route.js]: Why are you using a `PUT` / `PATCH` request to create a resource?

File: Server/routes/route.js [hunk 5]
```diff
@@ -1,46 +1,42 @@
...
+router.get('/parcels',validateToken, getParcelsByUser);
+// this is to enable a user change the dropoff location of a parcel
+router.patch('/parcels/:pid/destination',validateLocationUpdate, validateToken, updateParcelDestination);
 // this is to enable a user to cancel a parcel delivery order
-router.put('/parcels/:pid/cancel', validateToken, usersControllers.cancelParcelOrder);
-// this is to delete a specific parcel order
-// DELETE A PARCEL ORDER
-// router.delete('/parcels/:id', parcelController.deleteSpecificParcel);
-
+router.patch('/parcels/:pid/cancel', validateToken, cancelParcelOrder);
```
Comment (saheedt) [inline Server/routes/route.js]: No part of the rubrics instructed you to use `PATCH` request. It's rather unfortunate you don't pay attention to very simple details.

File: Server/routes/route.js [hunk 6]
```diff
@@ -1,46 +1,42 @@
...
 // this is the route to get all parcels for a user
-// GET ALL PARCELS
-router.get('/parcels/user', parcelController.getParcelsByUser);
-// this is the route to get a specific parcel
-// GET A SPECIFIC PARCEL
-// router.get('/parcels/:id', parcelController.getSpecificParcel);
-// this is to change the dropofflocation of a parcel
-// PUT IN A NEW DROPOFFLOCATION
-router.put('/parcels/:pid/newlocation', validateToken, validateAdmin, parcelController.updateParcelPresentLocation);
+router.get('/parcels',validateToken, getParcelsByUser);
```
Comment (saheedt) [inline Server/routes/route.js]: This is still wrong. This is not the route specified in the rubric to get all parcels belonging to a user.

File: Server/routes/route.js (hunk 3 above)
Comment (bobbylemm) [reply Server/routes/route.js]: yes the rubics says patch

Comment (bobbylemm) [reply Server/routes/route.js]: screenshot from 2018-11-22 17-50-31

Comment (saheedt) [reply Server/routes/route.js]: No it doesn't!!!
screen shot 2018-11-22 at 5 51 10 pm

**Comment by saheedt** [comment]
Consult the RIGHT rubrics and use the right routes. Once that is done, make sure to update your test as it will make your tests fail.

File: Server/routes/route.js (hunk 2 above)
Comment (bobbylemm) [reply Server/routes/route.js]: it has been fixed

File: Server/routes/route.js (hunk 3 above)
Comment (bobbylemm) [reply Server/routes/route.js]: it has been fixed

File: Server/routes/route.js (hunk 4 above)
Comment (bobbylemm) [reply Server/routes/route.js]: it has been fixed

File: Server/routes/route.js (hunk 5 above)
Comment (bobbylemm) [reply Server/routes/route.js]: it has been fixed

File: Server/routes/route.js (hunk 6 above)
Comment (bobbylemm) [reply Server/routes/route.js]: it has been fixed

**Pull request merged**